sdk/src/types/block/output/delegation.rs

```diff
@@ -179,18 +171,40 @@ impl DelegationOutputBuilder {
 
         let mut output = DelegationOutput {
             amount: 0,
-            delegated_amount: self.delegated_amount,
+            delegated_amount: 0,
             delegation_id: self.delegation_id,
             validator_address,
             start_epoch: self.start_epoch,
             end_epoch: self.end_epoch,
             unlock_conditions,
         };
 
-        output.amount = match self.amount {
-            OutputBuilderAmount::Amount(amount) => amount,
-            OutputBuilderAmount::MinimumAmount(params) => output.minimum_amount(params),
-        };
+        match self.delegated_amount {
+            OutputBuilderAmount::Amount(amount) => {
+                output.delegated_amount = amount;
+                output.amount = if let Some(builder_amount) = self.amount {
+                    match builder_amount {
+                        OutputBuilderAmount::Amount(amount) => amount,
+                        OutputBuilderAmount::MinimumAmount(params) => output.minimum_amount(params),
+                    }
+                } else {
+                    amount
+                };
+            }
+            OutputBuilderAmount::MinimumAmount(params) => {
+                let min = output.minimum_amount(params);
+                output.delegated_amount = min;
+                output.amount = if let Some(builder_amount) = self.amount {
+                    if let OutputBuilderAmount::Amount(amount) = builder_amount {
+                        amount
+                    } else {
```

Comment:
we allow the user to set it to something smaller than `min`?

Reply:
yeah, technically you can do this with all the outputs, but of course it will not be accepted